File: app/src/main/java/ru/serjik/utils/AssetsUtils.java
package ru.serjik.utils;

import android.content.res.AssetManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import java.io.Closeable;
import java.io.IOException;
import java.io.InputStream;

public class AssetsUtils {
    public static String readText(String path, AssetManager assets) {
        InputStream is = null;
        try {
            is = assets.open(path);
            return StreamUtils.toString(is);
        } catch (IOException e) {
            SerjikLog.log(e.getMessage());
            throw new RuntimeException(e);
        } finally {
            StreamUtils.closeQuietly(is);
        }
    }

    public static Bitmap readBitmap(String path, AssetManager assets) {
        InputStream is = null;
        try {
            is = assets.open(path);
            return BitmapFactory.decodeStream(is);
        } catch (IOException e) {
            SerjikLog.log(e.getMessage());
            throw new RuntimeException(e);
        } finally {
            StreamUtils.closeQuietly(is);
        }
    }
}

File: app/src/main/java/ru/serjik/utils/AssetsUtils.kt
package ru.serjik.utils

import android.content.res.AssetManager
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import java.io.IOException

object AssetsUtils {
    fun readText(path: String, assets: AssetManager): String {
        try {
            return assets.open(path).use { input ->
                StreamUtils.toString(input)
            }
        } catch (e: IOException) {
            SerjikLog.log(e.message ?: "")
            throw RuntimeException(e)
        }
    }

    fun readBitmap(path: String, assets: AssetManager): Bitmap {
        try {
            return assets.open(path).use { input ->
                BitmapFactory.decodeStream(input)
            }
        } catch (e: IOException) {
            SerjikLog.log(e.message ?: "")
            throw RuntimeException(e)
        }
    }
}
